### PR TITLE
ci: Fix ctr cannot delete a non stopped container

### LINKED ISSUE #3896: failures on CI s390 when running ksm test
Seems that the CI is failing with the ksm test
```
14:33:20 KSM stabilised at 9s
14:33:20 Pages merged 18358
14:33:20 Comparing merged pages between containers
14:33:20 time="2021-08-30T15:33:21-04:00" level=error msg="failed to delete container \"test\"" error="cannot delete a non stopped container: {running 0 0001-01-01 00:00:00 +0000 UTC}"
14:33:21 ctr: cannot delete a non stopped container: {running 0 0001-01-01 00:00:00 +0000 UTC}
14:33:21 make: *** [Makefile:113: ksm] Error 1
```
See http://jenkins.katacontainers.io/job/kata-containers-2.0-tests-ubuntu-s390x-PR/156/console
/cc @Jakob-Naucke @dborquez

### PR DESCRIPTION
**Pull request by dborquez**
For the function clean_env_ctr, adds a loop that waits
until all running processes are stopped before to proceed
to delete all running containers.

Fixes #3896

Signed-off-by: David Esparza <david.esparza.borquez@intel.com>

### CODE REVIEW

**Comment by cmaf** [comment]
@dborquez is the maximum based on testing?

**Comment by GabyCT** [comment]
@dborquez you have the following error
```
16:46:13 [default_vcpus_test.sh:186] ERROR: sudo ctr run --runtime=io.containerd.run.kata.v2 -d quay.io/prometheus/busybox:latest test sh -c nproc | grep 4
16:46:13 make: *** [Makefile:155: vcpus] Error 1
16:46:13 [run.sh:44] ERROR: sudo -E PATH=/home/jenkins/workspace/kata-containers-2.0-tests-ubuntu-s390x-PR/go/bin:/usr/local/go/bin:/usr/sbin:/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin CRI_RUNTIME=containerd bash -c make vcpus
```

File: lib/common.bash [hunk 1]
```diff
@@ -170,13 +170,25 @@ clean_env_ctr()
...
 
 	for i in "${containers[@]}"; do
-		sudo ctr tasks kill $(sudo ctr task ls | grep $i)
+		sudo ctr tasks kill "$i"
+	done
+
+
+
+	local j=0
+	while [ "$j" -lt "$time_out" ]; do
```
Comment (Jakob-Naucke) [inline lib/common.bash:185]: I should have looked at it before running the retest, but you could use https://github.com/kata-containers/tests/blob/435563e2dba252d9af217185f58dc52f77e9e75e/.ci/lib.sh#L173 here

**Comment by Jakob-Naucke** [comment]
just tested -- no backport needed

**Review by GabyCT — APPROVED** [review]

**Comment by dborquez** [comment]
@cmaf thank you:
> @dborquez is the maximum based on testing?
it was the maximum for 128 running containers

File: lib/common.bash [hunk 2]
```diff
@@ -170,14 +170,21 @@ clean_env_ctr()
...
 	local count_running="${#containers[@]}"
+	local count_stopped=0
+	local time_out=10
+	local sleep_time=1
 
 	[ "$count_running" -eq "0" ] && return 0
 
 	for i in "${containers[@]}"; do
-		sudo ctr tasks kill $(sudo ctr task ls | grep $i)
+		sudo ctr tasks kill "$i"
```
Comment (Jakob-Naucke) [inline lib/common.bash:180]: TIL this is possible

**Review by Jakob-Naucke — APPROVED** [review]

**Comment by devimc** [comment]
/test-ubuntu-metrics

lgtm - @wainersm wdyt ?

**Comment by Jakob-Naucke** [comment]
Hmm, there are still issues on s390x that seem related:
```
KSM stabilised at 4s
Pages merged 319
Comparing merged pages between containers
[ksm_test.sh:184] INFO: Wait until the containers gets removed
make: *** [Makefile:113: ksm] Error 1
[run.sh:52] ERROR: sudo -E PATH=/home/jenkins/workspace/kata-containers-2.0-tests-ubuntu-s390x-PR/go/bin:/usr/local/go/bin:/usr/sbin:/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin CRI_RUNTIME=containerd bash -c make ksm
[jenkins_job_build.sh:193] ERROR: .ci/run.sh
```

**Comment by GabyCT** [comment]
> Hmm, there are still issues on s390x that seem related:
> 
> ```
> KSM stabilised at 4s
> Pages merged 319
> Comparing merged pages between containers
> [ksm_test.sh:184] INFO: Wait until the containers gets removed
> make: *** [Makefile:113: ksm] Error 1
> [run.sh:52] ERROR: sudo -E PATH=/home/jenkins/workspace/kata-containers-2.0-tests-ubuntu-s390x-PR/go/bin:/usr/local/go/bin:/usr/sbin:/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin CRI_RUNTIME=containerd bash -c make ksm
> [jenkins_job_build.sh:193] ERROR: .ci/run.sh
> ```

^ @dborquez

**Comment by GabyCT** [comment]
> Hmm, there are still issues on s390x that seem related:
> 
> ```
> KSM stabilised at 4s
> Pages merged 319
> Comparing merged pages between containers
> [ksm_test.sh:184] INFO: Wait until the containers gets removed
> make: *** [Makefile:113: ksm] Error 1
> [run.sh:52] ERROR: sudo -E PATH=/home/jenkins/workspace/kata-containers-2.0-tests-ubuntu-s390x-PR/go/bin:/usr/local/go/bin:/usr/sbin:/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin CRI_RUNTIME=containerd bash -c make ksm
> [jenkins_job_build.sh:193] ERROR: .ci/run.sh
> ```

^ @dborquez

**Comment by wainersm** [comment]
> /test-ubuntu-metrics
> 
> lgtm - @wainersm wdyt ?

lgtm too. I will drop this -> https://github.com/kata-containers/tests/pull/3916

**Review by devimc — APPROVED** [review]
thanks @dborquez

**Comment by Jakob-Naucke** [comment]
> ```
> KSM stabilised at 4s
> Pages merged 319
> Comparing merged pages between containers
> [ksm_test.sh:184] INFO: Wait until the containers gets removed
> make: *** [Makefile:113: ksm] Error 1
> [run.sh:52] ERROR: sudo -E PATH=/home/jenkins/workspace/kata-containers-2.0-tests-ubuntu-s390x-PR/go/bin:/usr/local/go/bin:/usr/sbin:/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin CRI_RUNTIME=containerd bash -c make ksm
> [jenkins_job_build.sh:193] ERROR: .ci/run.sh
> ```

I could swear I saw that once more on a dev machine with this applied, but I've been loop-running it for over an hour now with no issues. We must see what happens.

**Pull request merged**